presto-main/src/main/java/com/facebook/presto/operator/AggregationOperator.java

```diff
@@ -256,4 +323,57 @@ public long getEstimatedSize()
             return aggregation.getEstimatedSize();
         }
     }
+
+    private static class AggregationOrderProcessor
+    {
+        private final List<Integer> orderByChannels;
+        private final List<SortOrder> orderings;
+        private final int[] orderByOutputChannels;
+        private final PagesIndex pageIndex;
+        private final PageBuilder pageBuilder;
+        private int currentPosition;
+
+        private AggregationOrderProcessor(
+                List<Type> orderBySourceTypes,
+                List<Integer> orderByChannels,
+                List<SortOrder> orderings,
+                PagesIndex.Factory pageIndexFactory)
+        {
+            this.orderByOutputChannels = new int[orderBySourceTypes.size()];
+            for (int i = 0; i < orderBySourceTypes.size(); i++) {
+                orderByOutputChannels[i] = i;
+            }
+            this.orderByChannels = orderByChannels;
+            this.orderings = orderings;
+            this.pageIndex = pageIndexFactory.newPagesIndex(orderBySourceTypes, 10_000);
+            this.pageBuilder = new PageBuilder(orderBySourceTypes);
```

Comment:
So keep orderBySourceTypes as a private member variable? Doesn't look that much better than this.